src/components/templates/Search/sort.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { ReactElement } from 'react'
+import { useNavigate } from '@reach/router'
+import {
+  addExistingParamsToUrl,
+  SortTermOptions,
+  SortValueOptions
+} from './utils'
+import Button from '../../atoms/Button'
+import styles from './sort.module.css'
+import classNames from 'classnames/bind'
+
+const cx = classNames.bind(styles)
+
+const sortItems = [
+  { display: 'Published', value: SortTermOptions.Created },
+  { display: 'Liquidity', value: SortTermOptions.Liquidity },
+  { display: 'Price', value: SortTermOptions.Price }
+]
+
+export default function Sort({
+  sortType,
+  setSortType,
+  sortDirection,
+  setSortDirection
+}: {
+  sortType: string
+  setSortType: React.Dispatch<React.SetStateAction<string>>
+  sortDirection: string
+  setSortDirection: React.Dispatch<React.SetStateAction<string>>
+}): ReactElement {
+  const navigate = useNavigate()
+  async function sortResults(sortBy?: string, direction?: string) {
+    let urlLocation: string
+    if (sortBy) {
+      urlLocation = await addExistingParamsToUrl(location, 'sort')
+      urlLocation = `${urlLocation}&sort=${sortBy}`
+    } else if (direction) {
+      urlLocation = await addExistingParamsToUrl(location, 'sortOrder')
+      urlLocation = `${urlLocation}&sortOrder=${direction}`
+    }
+    navigate(urlLocation)
+  }
+  return (
+    <div className={styles.sortList}>
+      <label className={styles.sortLabel}>Sort</label>
+      {sortItems.map((e, index) => {
+        const sorted = cx({
+          [styles.selected]: e.value === sortType,
+          [styles.sorted]: true
+        })
+        return (
+          <Button
+            key={index}
+            className={sorted}
+            size="small"
+            onClick={() => {
+              if (e.value === sortType) {
```

Comment:
a bit long so the justifies moving this into it's own `handleButtonClick` handler. Even more optimization could be to remove that double `return`, and have a `<SortItem />` component in same file abstracting out the single component from the `<Sort />` component

Reply:
You mean merging both Filter and Sort in the same file and returning everything within the same component ?

Reply:
ok, got it 😃

Reply:
uuuh, hang on, just looking into this and that might be actually too much refactor. So let's keep the double return but move out the `onClick()` stuff, then it's a good compromise